test/unit/abilities/forums_test.rb

```diff
@@ -6,51 +6,72 @@ module Abilities
   class ForumsTest < ActiveSupport::TestCase
     def setup
       @provider = FactoryBot.create(:provider_account)
-      @admin_user = @provider.admins.first
-      # admin = FactoryBot.create(:admin, account: @provider)
-      @provider_user = FactoryBot.create(:user, account: @provider)
+    end
 
-      @forum = @provider.forum
-      @topic = FactoryBot.create(:topic, forum: @forum)
+    class PublicForumTest < ForumsTest
+      def setup
+        super
```

Comment:
was not inner class already invoke outher class `#setup`? I'm not sure, just asking whether `@provider = FactoryBot.create(:provider_account)` would be called twice.

Reply:
Not sure if I understand the question but I'm positive provider is not instantiated twice